packages/utils/tests/getRandomInt.test.ts

```diff
@@ -0,0 +1,75 @@
+import { randomInt } from 'crypto';
+
+import { getRandomInt } from '../src/getRandomInt';
+
+describe(getRandomInt.name, () => {
+    it('raises same error as randomInt from crypto when max <= min', () => {
+        const EXPECTED_ERROR = new RangeError(
+            'The value of "max" is out of range. It must be greater than the value of "min" (0). Received -1',
+        );
+
+        expect(() => randomInt(0, -1)).toThrowError(EXPECTED_ERROR);
+        expect(() => getRandomInt(0, -1)).toThrowError(EXPECTED_ERROR);
+    });
+
+    it('raises error for range > 2^32', () => {
+        const EXPECTED_ERROR = new RangeError(
+            'This function only provide 32 bits of entropy, therefore range cannot be more then 2^32.',
+        );
+
+        expect(() => getRandomInt(0, Math.pow(2, 32) + 1)).toThrowError(EXPECTED_ERROR);
+    });
+
+    it('raises same error for unsafe integer', () => {
+        const UNSAFE_INTEGER = 2 ** 53;
+
+        const EXPECTED_ERROR_MIN = new RangeError(
+            'The "min" argument must be a safe integer. Received type number (9007199254740992)',
+        );
+
+        expect(() => randomInt(UNSAFE_INTEGER, UNSAFE_INTEGER + 1)).toThrowError(
+            EXPECTED_ERROR_MIN,
+        );
+        expect(() => getRandomInt(UNSAFE_INTEGER, UNSAFE_INTEGER + 1)).toThrowError(
+            EXPECTED_ERROR_MIN,
+        );
+
+        const EXPECTED_ERROR_MAX = new RangeError(
+            'The "max" argument must be a safe integer. Received type number (9007199254740992)',
+        );
+
+        expect(() => randomInt(0, UNSAFE_INTEGER)).toThrowError(EXPECTED_ERROR_MAX);
+        expect(() => getRandomInt(0, UNSAFE_INTEGER)).toThrowError(EXPECTED_ERROR_MAX);
+    });
+
+    // This test takes 100+seconds to run. It is very needed for development and debugging,
+    // but due to the time it takes, it is skipped in CI.
+    it.skip('return value in given range and in uniform distribution', () => {
```

Comment:
skiiiip

Reply:
The commentary explains it. I ran it locally, it passes and took 2 min, so it makes sense to skip.

Though maybe it's no harm to enable it - after all, github runners run parallel, and unit tests are not by far the slowest job, so slowing it down would not slow down CI as a whole.. But it could add up as people add more tests. FWIW I'm fine with it either way :shrug:

Reply:
Alternatively: lower the number of samples, if it would still have statistical significance. And developers can increase it locally as they wish..

Reply:
Yes, then you would have to increase tolerance. I tried playing around with samples vs. tolerance locally. But even when you have high number of samples, _or_ high tolerance, it still _can_ randomly fail, although with very tiny probability (negligible).
BUt itill, it just feels to me methodically incorrect to run an indeterministic test in CI, even when statistics are overwhelmingly on our side. So I'm fine with skipping it, but that's just my opinion :smile: